src/mono/wasm/runtime/startup.ts

```diff
@@ -471,6 +472,17 @@ export function bindings_lazy_init(): void {
 
     _create_primitive_converters();
 
+    if (!ENVIRONMENT_IS_PTHREAD) {
+        // Install our sync context so that async continuations will migrate back to the main thread automatically
```

Comment:
Couple of things:

1. I'm not sure if this method still exists after the various startup.ts refactorings.
2. Is this installing a sync context on every non-main pthread?  Why?
3. I think if we want to do work in every non-main pthread, we should be doing it in a new `    pthreads_worker.currentWorkerThreadEvents.addEventListener(pthreads_worker.dotnetPthreadCreated, (event) => ...)` callback.  (see `mono_wasm_pthread_worker_init`) we want to do work every time a pthread is starts (but again... is that what we want?) not every time a new worker is created by Emscripten (and inparticular before the worker has a pthread first assigned to it, it seems too early to be calling wasm code from the worker)

Reply:
Right now it only installs it on the main thread, and then the concurrent apis like task will capture the sync context and bring it with them to the threadpool threads, I think? I'm not sure whether I like that or not. I'm open to doing it any particular way.

Good point about workers vs pthreads, I'll see if it's reasonable to do it in that worker_init function.

Reply:
I think I misread what is happening. I saw `if (!ENVIRONMENT_IS_PTHREAD) {` and didn't see the `!`. sorry.